engine/gamesys/src/gamesys/components/comp_mesh.cpp

```diff
@@ -729,6 +791,53 @@ namespace dmGameSystem
         }
     }
 
+    static void RenderListFrustumCulling(dmRender::RenderListVisibilityParams const &params)
+    {
+        DM_PROFILE("Mesh"); // TODO - use a proper name here
+
+        MeshWorld* mesh_world = (MeshWorld*)params.m_UserData;
+
+
+        const dmIntersection::Frustum frustum = *params.m_Frustum;
+        uint32_t num_entries = params.m_NumEntries;
+        for (uint32_t i = 0; i < num_entries; ++i)
+        {
+            dmRender::RenderListEntry* entry = &params.m_Entries[i];
+            MeshComponent* component_p = (MeshComponent*)entry->m_UserData;
+
+            dmGameSystem::BufferResource* br = GetVerticesBuffer(component_p, component_p->m_Resource);
+            dmVMath::Point3 boundsMin, boundsMax;
+            dmBuffer::Result r = dmBuffer::GetBounds(br->m_Buffer, boundsMin, boundsMax);
+            if (r != dmBuffer::RESULT_OK) {
+                dmLogWarning("Could not get culling bounds for mesh buffer. Is the buffer valid ?");
+                continue;
+            }
+
+            // are there any actual bounds set ? (checking if boundxMin still contains its initial values should be enough)
+            if (boundsMin.getX() == BOUNDS_MIN_EMPTY.getX() && boundsMin.getY() == BOUNDS_MIN_EMPTY.getY() && boundsMin.getZ() == BOUNDS_MIN_EMPTY.getZ()) {
+                continue;
+            }
```

Comment:
I wonder if this may be more "generic":

```c++
bool IsAABBValid(P3 min, P3 max) {
    return !(min.getX() > max.getX() ||
        min.getY() > max.getY() ||
        min.getZ() > max.getZ());
}
```

I like the fact that the calculation is "contained" and uses it's own local variables, rather than loading the global constant from memory.

Also, I think it's easier to communicate as an "API" to our users: "If the bounding box is invalid, the object will not be culled."

Reply:
Both arguments stand. I like the suggestion too.

My only concert is whether we can assert that buffer bounds have been initialized if we choose this path. It can easily happen to have an invalid uninitialized bounding box with bound_max > bound_min. We'll take this a a valid one which is wrong. 

In the existing implementation it is highly improbable to have bounds set to +-1e10 out of bad luck. So, at least we won't worry about this.

Reply:
i would place this in dlib/buffer.h. Make it private for now.

Reply:
It's not really a buffer thing but since the bounds are initialized in buffer.cpp, it seems ok to me.